src/hooks/useTodoListItemCreate.tsx

```diff
@@ -0,0 +1,26 @@
+import uuid4 from 'uuid4';
+import { postTodoListItem } from '@todo/api';
+import { TodoItem } from '@todo/store';
+import { useTodoList } from '.';
+
+const useTodoListItemCreate = () => {
+  const { queryTodoList } = useTodoList();
+  const todoListItemCreate = async (todoItem: TodoItem, dueDate: string) => {
+    todoItem.id = uuid4();
+    todoItem.dueDate = dueDate;
+    todoItem.cardColor = +todoItem.cardColor;
+    todoItem.category = +todoItem.category;
+
+    try {
+      await postTodoListItem(todoItem);
+    } catch (error) {
+      console.log(error);
+    } finally {
+      queryTodoList();
+    }
+  };
+
+  return { todoListItemCreate };
+};
+
+export { useTodoListItemCreate };
```

Comment:
You created a hook with only one method inside, you could do it inside the component itself.

Reply:
Component includes Add and Edit methods inside. If I move inside than Edit in the feature component will bloat.